### PR TITLE
remove page reload calls and add callbacks

### PR DESCRIPTION
**Pull request by leem42**
Added callback functions to synapse client login/sign out functions
Used Universal-Cookie library to replace cookie functions in synapse client
Made it so the portal-side login modal shows up from synapsetable when clicking download files, this is the only way I could have the session callback for the portals get passed in without using a global variable.

### CODE REVIEW

**Comment by jay-hodgson** [comment]
Prefer to surface encoded query request bundle as an url query param

File: src/demo/containers/App.tsx [hunk 1]
```diff
@@ -65,6 +74,7 @@ export default class App extends React.Component<{}, AppState> {
           token={
             SynapseClient.IS_OUTSIDE_SYNAPSE_ORG ? token : this.state.token
```
Comment (leem42) [inline src/demo/containers/App.tsx]: document on yarn link react in portals

File: src/lib/utils/SynapseClient.ts [hunk 2]
```diff
@@ -955,35 +958,52 @@ export const getWikiAttachmentsFromEvaluation = (
...
+    if (!token) {
+      cookies.remove(SESSION_TOKEN_COOKIE_KEY)
+    } else {
+      // set's cookie in session storage
+      cookies.set(SESSION_TOKEN_COOKIE_KEY, token, {
+        // expires in a day
+        maxAge: 60 * 60 * 24,
+      })
+    }
+    sessionCallback?.()
```
Comment (leem42) [inline src/lib/utils/SynapseClient.ts]: make mandatory

File: src/lib/utils/SynapseClient.ts [hunk 3]
```diff
@@ -955,35 +958,52 @@ export const getWikiAttachmentsFromEvaluation = (
...
+    return
```
Comment (leem42) [inline src/lib/utils/SynapseClient.ts]: return void

File: src/lib/utils/SynapseClient.ts [hunk 4]
```diff
@@ -1049,20 +1069,10 @@ export const detectSSOCode = () => {
...
-        console.error('err when clearing the session cookie ', err)
-      })
-  } else {
-    // is not on a synapse.org subdomain
-    removeCookie(SESSION_TOKEN_COOKIE_KEY)
-    window.location.reload()
-  }
+export const signOut = (sessionCallback?: Function) => {
+  setSessionTokenCookie(undefined, sessionCallback).catch(err => {
+    console.error('err when clearing the session cookie ', err)
```
Comment (leem42) [inline src/lib/utils/SynapseClient.ts]: nothing to catch, remove

Comment (leem42) [reply src/lib/utils/SynapseClient.ts]: make callback mandatory

**Pull request merged**